packages/core-data/src/index.js

```diff
@@ -76,3 +76,4 @@ register( store );
 export { default as EntityProvider } from './entity-provider';
 export * from './entity-provider';
 export * from './fetch';
+export * from './types';
```

Comment:
Maybe here it should export with `type` keyword?
```suggestion
export type * from './types';
```

Reply:
I guess I'm wrong since it might be either types or interfaces. There are more places where export isn't as specific as in other places that might create challenges for webpack.

Reply:
I think interfaces still fall under the `export type`/`import type` directives. Like types they're free to be erased by whatever is stripping out the TS annotations.

Reply:
I don't know good enough how this TypeScript setup works, but other packages that ship types to npm are listed in this file:

https://github.com/WordPress/gutenberg/blob/trunk/tsconfig.json

I don't see types exported from the entry point of other packages listed there, so maybe this is something you should look into.

Reply:
That keyword won't work in a .js file, but I[ updated the `index.ts` to say `type` next to every single import and export](https://github.com/WordPress/gutenberg/pull/38666/commits/b23a08a40da9a801dba84d3296fbac22dae1a929) and now it's all good – I must have done something wrong yesterday 👍